src/legacy/core_plugins/kibana/public/management/sections/settings/components/component_registry.ts

```diff
@@ -17,7 +17,10 @@
  * under the License.
  */
 
-const registry = {};
+import { FunctionComponent } from 'react';
+
+type Id = string;
+const registry: Record<Id, FunctionComponent> = {};
```

Comment:
```suggestion
const registry: Record<Id, React.ComponentType> = {};
```

`FunctionComponent` is type of an instance of a component, I assume you want to use component type itself here.